test/functional/feature_poolpair_fork.py

```diff
@@ -0,0 +1,219 @@
+        self.setup_clean_chain = True
+        self.extra_args = [
+            ['-txnotokens=0', '-amkheight=50', '-bayfrontheight=50', '-dakotaheight=300'],
+            ['-txnotokens=0', '-amkheight=50', '-bayfrontheight=50', '-dakotaheight=300'],
+            ['-txnotokens=0', '-amkheight=50', '-bayfrontheight=50', '-dakotaheight=300'],
+            ['-txnotokens=0', '-amkheight=50', '-bayfrontheight=50', '-dakotaheight=300']]
+
+
+    def run_test(self):
+        assert_equal(len(self.nodes[0].listtokens()), 1) # only one token == DFI
+
+        self.nodes[0].generate(100)
+        self.sync_all()
+
+        self.nodes[2].generate(100)
+        self.sync_all()
+
+        # Stop node #3 for future revert
+        self.stop_node(3)
+
+        # CREATION:
+        #========================
+        collateral0 = self.nodes[0].getnewaddress("", "legacy")
+
+        self.nodes[0].generate(1)
+
+        # 1 Creating DAT token
+        self.nodes[0].createtoken({
+            "symbol": "PT",
+            "name": "Platinum",
+            "isDAT": True,
+            "collateralAddress": collateral0
+        })
+
+        self.nodes[0].generate(1)
+        self.sync_blocks([self.nodes[0], self.nodes[2]])
+
+        # At this point, token was created
+        tokens = self.nodes[0].listtokens()
+        assert_equal(len(tokens), 2)
+        assert_equal(tokens['1']["symbol"], "PT")
+
+        # check sync:
+        tokens = self.nodes[2].listtokens()
+        assert_equal(len(tokens), 2)
+        assert_equal(tokens['1']["symbol"], "PT")
+
+        # 3 Trying to make regular token
+        self.nodes[0].generate(1)
+        createTokenTx = self.nodes[0].createtoken({
+            "symbol": "GOLD",
+            "name": "shiny gold",
+            "isDAT": False,
+            "collateralAddress": collateral0
+        })
+        self.nodes[0].generate(1)
+        # Checks
+        tokens = self.nodes[0].listtokens()
+        assert_equal(len(tokens), 3)
+        assert_equal(tokens['128']["symbol"], "GOLD")
+        assert_equal(tokens['128']["creationTx"], createTokenTx)
+
+        # 7 Creating PoolPair from Foundation -> OK
+        self.nodes[0].createpoolpair({
+            "tokenA": "PT",
+            "tokenB": "GOLD#128",
+            "comission": 0.001,
+            "status": True,
+            "ownerAddress": collateral0,
+            "pairSymbol": "PTGOLD"
+        }, [])
+
+        self.nodes[0].generate(1)
+        # Trying to create the same again and fail
+        try:
+            self.nodes[0].createpoolpair({
+            "tokenA": "PT",
+            "tokenB": "GOLD#128",
+            "comission": 0.001,
+            "status": True,
+            "ownerAddress": collateral0,
+            "pairSymbol": "PTGD"
+        }, [])
+        except JSONRPCException as e:
+            errorString = e.error['message']
+        assert("Error, there is already a poolpairwith same tokens, but different poolId" in errorString)
```

Comment:
Can we use `assert_raises_rpc_error` instead of catching code all over the field. I know other dev use it, but probably they didn't know about.